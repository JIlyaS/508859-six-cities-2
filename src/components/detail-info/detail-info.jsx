import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import ReviewsList from '../reviews-list/reviews-list';
import Map from '../map/map';
import OfferList from '../offer-list/offer-list';
import {convertRating} from '../../utils';
import {MAX_NEARBY_OFFER} from '../../constants';

const DetailInfo = (props) => {
  const {
    currentOffer: {isPremium, price, title, rating, photos, features, insideProperties, hostUser, reviews},
    otherOffers
  } = props;
  const nearByCoordinates = otherOffers.map((offer) => offer.coordinate);
  return <div className="page">
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <a className="header__logo-link" href="/">
              <img className="header__logo" src="/img/logo.svg" alt="6 cities logo" width="81" height="41" />
            </a>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                <a className="header__nav-link header__nav-link--profile" href="#">
                  <div className="header__avatar-wrapper user__avatar-wrapper">
                  </div>
                  <span className="header__user-name user__name">Oliver.conner@gmail.com</span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>

    <main className="page__main page__main--property">
      <section className="property">
        <div className="property__gallery-container container">
          <div className="property__gallery">
            {photos.map((photo, id) => <div className="property__image-wrapper" key={`photo-${id}`}>
              <img className="property__image" src={`/img/${photo}`} alt="Photo studio" />
            </div>)}
          </div>
        </div>
        <div className="property__container container">
          <div className="property__wrapper">
            {isPremium && <div className="property__mark">
              <span>Premium</span>
            </div>}
            <div className="property__name-wrapper">
              <h1 className="property__name">
                {title}
              </h1>
              <button className="property__bookmark-button button" type="button">
                <svg className="property__bookmark-icon" width="31" height="33">
                  <use xlinkHref="#icon-bookmark"></use>
                </svg>
                <span className="visually-hidden">To bookmarks</span>
              </button>
            </div>
            <div className="property__rating rating">
              <div className="property__stars rating__stars">
                <span style={{width: convertRating(rating) + `%`}}></span>
                <span className="visually-hidden">Rating</span>
              </div>
              <span className="property__rating-value rating__value">{rating}</span>
            </div>
            <ul className="property__features">
              {features.map((feature, id) => <li className="property__feature property__feature--entire" key={`feature-${id}`}>
                {feature}
              </li>)}
            </ul>
            <div className="property__price">
              <b className="property__price-value">&euro;{price}</b>
              <span className="property__price-text">&nbsp;night</span>
            </div>
            <div className="property__inside">
              <h2 className="property__inside-title">What&apos;s inside</h2>
              <ul className="property__inside-list">
                {insideProperties.map((insideProperty, id) => <li className="property__inside-item" key={`inside-property-${id}`}>
                  {insideProperty}
                </li>)}
              </ul>
            </div>
            <div className="property__host">
              <h2 className="property__host-title">Meet the host</h2>
              <div className="property__host-user user">
                <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                  <img className="property__avatar user__avatar" src={`/img/${hostUser.avatar}`} width="74" height="74" alt="Host avatar" />
                </div>
                <span className="property__user-name">
                  {hostUser.name}
                </span>
                <span className="property__user-status">
                  {hostUser.status}
                </span>
              </div>
              <div className="property__description">
                {hostUser.description.map((descriptionItem, id) => <p className="property__text" key={`text-${id}`}>
                  {descriptionItem}
                </p>)}
              </div>
            </div>
            <section className="property__reviews reviews">
              <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length}</span></h2>
              <ReviewsList reviews={reviews} />
              <form className="reviews__form form" action="#" method="post">
                <label className="reviews__label form__label" htmlFor="review">Your review</label>
                <div className="reviews__rating-form form__rating">
                  <input className="form__rating-input visually-hidden" name="rating" value="5" id="5-stars" type="radio" />
                  <label htmlFor="5-stars" className="reviews__rating-label form__rating-label" title="perfect">
                    <svg className="form__star-image" width="37" height="33">
                      <use xlinkHref="#icon-star"></use>
                    </svg>
                  </label>

                  <input className="form__rating-input visually-hidden" name="rating" value="4" id="4-stars" type="radio" />
                  <label htmlFor="4-stars" className="reviews__rating-label form__rating-label" title="good">
                    <svg className="form__star-image" width="37" height="33">
                      <use xlinkHref="#icon-star"></use>
                    </svg>
                  </label>

                  <input className="form__rating-input visually-hidden" name="rating" value="3" id="3-stars" type="radio" />
                  <label htmlFor="3-stars" className="reviews__rating-label form__rating-label" title="not bad">
                    <svg className="form__star-image" width="37" height="33">
                      <use xlinkHref="#icon-star"></use>
                    </svg>
                  </label>

                  <input className="form__rating-input visually-hidden" name="rating" value="2" id="2-stars" type="radio" />
                  <label htmlFor="2-stars" className="reviews__rating-label form__rating-label" title="badly">
                    <svg className="form__star-image" width="37" height="33">
                      <use xlinkHref="#icon-star"></use>
                    </svg>
                  </label>

                  <input className="form__rating-input visually-hidden" name="rating" value="1" id="1-star" type="radio" />
                  <label htmlFor="1-star" className="reviews__rating-label form__rating-label" title="terribly">
                    <svg className="form__star-image" width="37" height="33">
                      <use xlinkHref="#icon-star"></use>
                    </svg>
                  </label>
                </div>
                <textarea className="reviews__textarea form__textarea" id="review" name="review" placeholder="Tell how was your stay, what you like and what can be improved"></textarea>
                <div className="reviews__button-wrapper">
                  <p className="reviews__help">
                    To submit review please make sure to set <span className="reviews__star">rating</span> and describe your stay with at least <b className="reviews__text-amount">50 characters</b>.
                  </p>
                  <button className="reviews__submit form__submit button" type="submit" disabled="">Submit</button>
                </div>
              </form>
            </section>
          </div>
        </div>
        <section className="property__map map">
          <Map coordinates={[...nearByCoordinates]} />
        </section>
      </section>
      <div className="container">
        <section className="near-places places">
          <h2 className="near-places__title">Other places in the neighbourhood</h2>
          <OfferList offers={otherOffers} isNearPlace />
        </section>
      </div>
    </main>
  </div>;
};

DetailInfo.propTypes = {
  idPath: PropTypes.string.isRequired,
  allOffers: PropTypes.array.isRequired
};

DetailInfo.propTypes = {
  currentOffer: PropTypes.shape({
    isPremium: PropTypes.bool.isRequired,
    img: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    type: PropTypes.oneOf([`apartment`, `private room`, `house`, `hotel`]),
    photos: PropTypes.array.isRequired,
    features: PropTypes.array.isRequired,
    insideProperties: PropTypes.array.isRequired,
    hostUser: PropTypes.shape({
      avatar: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      status: PropTypes.string.isRequired,
      description: PropTypes.array.isRequired
    }),
    reviews: PropTypes.array.isRequired
  }),
  otherOffers: PropTypes.array.isRequired
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  currentOffer: state.allOffers.find((offer) => offer.id === `id${Number(ownProps.idPath)}`),
  otherOffers: state.allOffers.filter((offer) => offer.id !== `id${Number(ownProps.idPath)}`).slice(0, MAX_NEARBY_OFFER)
});

export {DetailInfo};

export default connect(mapStateToProps)(DetailInfo);
