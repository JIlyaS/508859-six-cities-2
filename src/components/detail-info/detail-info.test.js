import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {DetailInfo} from './detail-info';
import {
  CURRENT_OFFER,
  DEFAULT_OFFERS,
  DEFAULT_OFFER,
  DEFAULT_REVIEW,
  DEFAULT_LOGIN,
  DetailInfoMock
} from '../../constants';

Enzyme.configure({adapter: new Adapter()});

jest.mock(`../map/map`);

it(`DetailInfo correctly renders after relaunch`, () => {
  const tree = shallow(<DetailInfo
    match={{params: {
      offerId: DetailInfoMock.ID
    }}}
    currentOffer={CURRENT_OFFER}
    otherOffers={DEFAULT_OFFERS}
    activeOfferCard={DEFAULT_OFFER}
    offers={DEFAULT_OFFERS}
    reviews={DEFAULT_REVIEW}
    idPath={DetailInfoMock.ID}
    login={DEFAULT_LOGIN}
    onGetLogin={() => {}}
    onLoadReviews={() => {}}
    onChangeOfferFavorite={() => {}}
  />);

  expect(toJSON(tree)).toMatchSnapshot();
});
