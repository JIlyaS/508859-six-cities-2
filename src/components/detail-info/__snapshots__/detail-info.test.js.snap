// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailInfo correctly renders after relaunch 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
            key="photo-0"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/room.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
            key="photo-1"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
            key="photo-2"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
            key="photo-3"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/apartment-03.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
            key="photo-4"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/studio-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
            key="photo-5"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/apartment-01.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Beautiful & luxurious apartment at great location
            </h1>
            <button
              className="property__bookmark-button button"
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "96%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4.8
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
              key="feature-0"
            >
              Entire place
            </li>
            <li
              className="property__feature property__feature--entire"
              key="feature-1"
            >
              3 Bedrooms
            </li>
            <li
              className="property__feature property__feature--entire"
              key="feature-2"
            >
              Max 4 adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              120
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
                key="inside-property-0"
              >
                Wi-Fi
              </li>
              <li
                className="property__inside-item"
                key="inside-property-1"
              >
                Washing machine
              </li>
              <li
                className="property__inside-item"
                key="inside-property-2"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
                key="inside-property-3"
              >
                Heating
              </li>
              <li
                className="property__inside-item"
                key="inside-property-4"
              >
                Coffee machine
              </li>
              <li
                className="property__inside-item"
                key="inside-property-5"
              >
                Baby seat
              </li>
              <li
                className="property__inside-item"
                key="inside-property-6"
              >
                Kitchen
              </li>
              <li
                className="property__inside-item"
                key="inside-property-7"
              >
                Dishwasher
              </li>
              <li
                className="property__inside-item"
                key="inside-property-8"
              >
                Cabel TV
              </li>
              <li
                className="property__inside-item"
                key="inside-property-9"
              >
                Fridge
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="/img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
              <span
                className="property__user-status"
              >
                pro
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
                key="text-0"
              >
                A quiet cozy and picturesque that hides behind a a river
      by the unique lightness of Amsterdam. The building is green
      and from 18th century.
              </p>
              <p
                className="property__text"
                key="text-1"
              >
                An independent House, strategically
      located between Rembrand Square and National Opera, but where
      the bustle of the city comes to rest in this alley flowery and
      colorful.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                2
              </span>
            </h2>
            <ReviewsList
              reviews={
                Array [
                  Object {
                    "avatar": "avatar-max.jpg",
                    "date": "April 2019",
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "id": "review1",
                    "name": "Max",
                    "rating": 4.7,
                  },
                  Object {
                    "avatar": "avatar-max.jpg",
                    "date": "May 2019",
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "id": "review2",
                    "name": "Alex",
                    "rating": 4,
                  },
                ]
              }
            />
            <form
              action="#"
              className="reviews__form form"
              method="post"
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  className="form__rating-input visually-hidden"
                  id="5-stars"
                  name="rating"
                  type="radio"
                  value="5"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="4-stars"
                  name="rating"
                  type="radio"
                  value="4"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="good"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="3-stars"
                  name="rating"
                  type="radio"
                  value="3"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="not bad"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="2-stars"
                  name="rating"
                  type="radio"
                  value="2"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="badly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="1-star"
                  name="rating"
                  type="radio"
                  value="1"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-star"
                  title="terribly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                id="review"
                name="review"
                placeholder="Tell how was your stay, what you like and what can be improved"
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set 
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled=""
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      >
        <Map
          activeCoordinate={
            Array [
              52.3909553943508,
              4.85309666406198,
            ]
          }
          coordinates={
            Array [
              Array [
                52.369553943508,
                4.85309666406198,
              ],
              Array [
                52.3909553943508,
                4.929309666406198,
              ],
              Array [
                52.3809553943508,
                4.939309666406198,
              ],
            ]
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <Connect(OfferList)
          isNearPlace={true}
          offers={
            Array [
              Object {
                "city": Object {
                  "location": Object {
                    "latitude": 52.3909553943508,
                    "longitude": 4.85309666406198,
                    "zoom": 12,
                  },
                  "name": "Amsterdam",
                },
                "coordinate": Array [
                  52.3909553943508,
                  4.85309666406198,
                ],
                "features": Array [
                  "Entire place",
                  "3 Bedrooms",
                  "Max 4 adults",
                ],
                "hostUser": Object {
                  "avatar": "avatar-angelina.jpg",
                  "description": Array [
                    "A quiet cozy and picturesque that hides behind a a river
      by the unique lightness of Amsterdam. The building is green
      and from 18th century.",
                    "An independent House, strategically
      located between Rembrand Square and National Opera, but where
      the bustle of the city comes to rest in this alley flowery and
      colorful.",
                  ],
                  "name": "Angelina",
                  "status": "pro",
                },
                "id": "id0",
                "img": "apartment-01.jpg",
                "insideProperties": Array [
                  "Wi-Fi",
                  "Washing machine",
                  "Towels",
                  "Heating",
                  "Coffee machine",
                  "Baby seat",
                  "Kitchen",
                  "Dishwasher",
                  "Cabel TV",
                  "Fridge",
                ],
                "isPremium": true,
                "photos": Array [
                  "room.jpg",
                  "apartment-01.jpg",
                  "apartment-02.jpg",
                  "apartment-03.jpg",
                  "studio-01.jpg",
                  "apartment-01.jpg",
                ],
                "price": 110,
                "rating": 4.8,
                "reviews": Array [
                  Object {
                    "avatar": "avatar-max.jpg",
                    "date": "April 2019",
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "id": "review1",
                    "name": "Max",
                    "rating": 4.7,
                  },
                  Object {
                    "avatar": "avatar-max.jpg",
                    "date": "May 2019",
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "id": "review2",
                    "name": "Alex",
                    "rating": 4,
                  },
                ],
                "title": "Beautiful & luxurious apartment at great location",
                "type": "apartment",
              },
              Object {
                "city": Object {
                  "location": Object {
                    "latitude": 52.3909553943508,
                    "longitude": 4.85309666406198,
                    "zoom": 12,
                  },
                  "name": "Amsterdam",
                },
                "coordinate": Array [
                  52.369553943508,
                  4.85309666406198,
                ],
                "features": Array [
                  "Entire place",
                  "3 Bedrooms",
                  "Max 4 adults",
                ],
                "hostUser": Object {
                  "avatar": "avatar-angelina.jpg",
                  "description": Array [
                    "A quiet cozy and picturesque that hides behind a a river
      by the unique lightness of Amsterdam. The building is green
      and from 18th century.",
                    "An independent House, strategically
      located between Rembrand Square and National Opera, but where
      the bustle of the city comes to rest in this alley flowery and
      colorful.",
                  ],
                  "name": "Angelina",
                  "status": "pro",
                },
                "id": "id1",
                "img": "room.jpg",
                "insideProperties": Array [
                  "Wi-Fi",
                  "Washing machine",
                  "Towels",
                  "Heating",
                  "Coffee machine",
                  "Baby seat",
                  "Kitchen",
                  "Dishwasher",
                  "Cabel TV",
                  "Fridge",
                ],
                "isPremium": false,
                "photos": Array [
                  "room.jpg",
                  "apartment-01.jpg",
                  "apartment-02.jpg",
                  "apartment-03.jpg",
                  "studio-01.jpg",
                  "apartment-01.jpg",
                ],
                "price": 120,
                "rating": 4,
                "reviews": Array [
                  Object {
                    "avatar": "avatar-max.jpg",
                    "date": "April 2019",
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "id": "review1",
                    "name": "Max",
                    "rating": 4.7,
                  },
                ],
                "title": "Wood and stone place",
                "type": "private room",
              },
              Object {
                "city": Object {
                  "location": Object {
                    "latitude": 52.3909553943508,
                    "longitude": 4.85309666406198,
                    "zoom": 12,
                  },
                  "name": "Amsterdam",
                },
                "coordinate": Array [
                  52.3909553943508,
                  4.929309666406198,
                ],
                "features": Array [
                  "Entire place",
                  "3 Bedrooms",
                  "Max 4 adults",
                ],
                "hostUser": Object {
                  "avatar": "avatar-angelina.jpg",
                  "description": Array [
                    "A quiet cozy and picturesque that hides behind a a river
      by the unique lightness of Amsterdam. The building is green
      and from 18th century.",
                    "An independent House, strategically
      located between Rembrand Square and National Opera, but where
      the bustle of the city comes to rest in this alley flowery and
      colorful.",
                  ],
                  "name": "Angelina",
                  "status": "pro",
                },
                "id": "id2",
                "img": "apartment-02.jpg",
                "insideProperties": Array [
                  "Wi-Fi",
                  "Washing machine",
                  "Towels",
                  "Heating",
                  "Coffee machine",
                  "Baby seat",
                  "Kitchen",
                  "Dishwasher",
                  "Cabel TV",
                  "Fridge",
                ],
                "isPremium": false,
                "photos": Array [
                  "room.jpg",
                  "apartment-01.jpg",
                  "apartment-02.jpg",
                  "apartment-03.jpg",
                  "studio-01.jpg",
                  "apartment-01.jpg",
                ],
                "price": 200,
                "rating": 4,
                "reviews": Array [
                  Object {
                    "avatar": "avatar-max.jpg",
                    "date": "April 2019",
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "id": "review1",
                    "name": "Max",
                    "rating": 4.7,
                  },
                ],
                "title": "Canal View Prinsengracht",
                "type": "apartment",
              },
              Object {
                "city": Object {
                  "location": Object {
                    "latitude": 52.3909553943508,
                    "longitude": 4.85309666406198,
                    "zoom": 12,
                  },
                  "name": "Amsterdam",
                },
                "coordinate": Array [
                  52.3809553943508,
                  4.939309666406198,
                ],
                "features": Array [
                  "Entire place",
                  "3 Bedrooms",
                  "Max 4 adults",
                ],
                "hostUser": Object {
                  "avatar": "avatar-angelina.jpg",
                  "description": Array [
                    "A quiet cozy and picturesque that hides behind a a river
      by the unique lightness of Amsterdam. The building is green
      and from 18th century.",
                    "An independent House, strategically
      located between Rembrand Square and National Opera, but where
      the bustle of the city comes to rest in this alley flowery and
      colorful.",
                  ],
                  "name": "Angelina",
                  "status": "pro",
                },
                "id": "id3",
                "img": "apartment-03.jpg",
                "insideProperties": Array [
                  "Wi-Fi",
                  "Washing machine",
                  "Towels",
                  "Heating",
                  "Coffee machine",
                  "Baby seat",
                  "Kitchen",
                  "Dishwasher",
                  "Cabel TV",
                  "Fridge",
                ],
                "isPremium": true,
                "photos": Array [
                  "room.jpg",
                  "apartment-01.jpg",
                  "apartment-02.jpg",
                  "apartment-03.jpg",
                  "studio-01.jpg",
                  "apartment-01.jpg",
                ],
                "price": 180,
                "rating": 5,
                "reviews": Array [
                  Object {
                    "avatar": "avatar-max.jpg",
                    "date": "April 2019",
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "id": "review1",
                    "name": "Max",
                    "rating": 4.7,
                  },
                ],
                "title": "Nice, cozy, warm big bed apartment",
                "type": "apartment",
              },
            ]
          }
        />
      </section>
    </div>
  </main>
</div>
`;
